cloudferry_devlab/cloudferry_devlab/tests/base.py

```diff
@@ -426,20 +426,35 @@ def set_vm_state(novaclient, vm_id, vm_state):
             return vm_id, 'ERROR'
         # emulate suspend state:
         elif vm_state == u'suspend':
-            novaclient.servers.suspend(vm_id)
+            try:
+                novaclient.servers.suspend(vm_id)
+            except nova_exceptions.Conflict:
+                pass
             return vm_id, 'SUSPENDED'
         # emulate resize state:
         elif vm_state == u'pause':
-            novaclient.servers.pause(vm_id)
+            try:
+                novaclient.servers.pause(vm_id)
+            except nova_exceptions.Conflict:
+                pass
             return vm_id, 'PAUSED'
         # emulate stop/shutoff state:
         elif vm_state == u'stop':
-            novaclient.servers.stop(vm_id)
+            try:
+                novaclient.servers.stop(vm_id)
+            except (nova_exceptions.Conflict, nova_exceptions.BadRequest):
+                pass
             return vm_id, 'SHUTOFF'
         # emulate resize state:
         elif vm_state == u'resize':
-            novaclient.servers.resize(vm_id, '2')
+            try:
+                novaclient.servers.resize(vm_id, '2')
+            except nova_exceptions.Conflict:
+                pass
             return vm_id, ('VERIFY_RESIZE', 'ACTIVE', 'ERROR')
+        elif vm_state == u'active':
+            novaclient.servers.reset_state(server=vm_id, state=vm_state)
```

Comment:
Did you try to run functional tests against VMs started like this? Do they work?